cookie-clicker-lab4-7/src/main/java/edu/csula/web/AdminGeneratorsServlet.java

```diff
@@ -0,0 +1,139 @@
+package edu.csula.web;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.Collection;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import edu.csula.models.Generator;
+import edu.csula.storage.GeneratorsDAO;
+import edu.csula.storage.UsersDAO;
+import edu.csula.storage.servlet.GeneratorsDAOImpl;
+import edu.csula.storage.servlet.UsersDAOImpl;
+
+@WebServlet("/admin/generators")
+public class AdminGeneratorsServlet extends HttpServlet {
+	@Override
+	public void doGet( HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+		response.setContentType("text/html");
+		PrintWriter out = response.getWriter();
+		// TODO: render the generators page HTML
+		//out.println("<h1>Hello generators servlet!</h1>");
+		GeneratorsDAO dao = new GeneratorsDAOImpl(getServletContext());
+		Collection<Generator> generators = dao.getAll();
```

Comment:
should probably still set the `generators` to the request scope.

This still work when the data is stored in the _application scope_ but will soon not work when we migrate the data from the application scope to database.